beacon_node/lighthouse_network/src/peer_manager/peerdb/peer_info.rs

```diff
@@ -362,25 +363,20 @@ impl<E: EthSpec> PeerInfo<E> {
         &mut self,
         meta_data: MetaData<E>,
         node_id_opt: Option<NodeId>,
+        peer_id: &PeerId,
         spec: &ChainSpec,
+        log: &Logger,
     ) {
         // If we don't have a node id, we cannot compute the custody duties anyway
         let Some(node_id) = node_id_opt else {
             self.meta_data = Some(meta_data);
             return;
         };
 
-        // Already set by enr if custody_subnets is non empty
-        if self.custody_subnets.is_empty() {
-            if let Ok(custody_subnet_count) = meta_data.custody_subnet_count() {
-                let custody_subnets = DataColumnSubnetId::compute_custody_subnets::<E>(
-                    node_id.raw(),
-                    std::cmp::min(*custody_subnet_count, spec.data_column_sidecar_subnet_count),
-                    spec,
-                )
-                .collect::<HashSet<_>>();
-                self.set_custody_subnets(custody_subnets);
-            }
+        if spec.is_peer_das_scheduled() {
```

Comment:
Even if peerdas is scheduled, they may not support metadata/v3, so on the reqresp protocol negotiation we will issue a metadata/v2. Can this lead to a lot of error logs when users progressively update to a peerdas version?

We can track how many of our peers are "peerdas ready", and gracefully ignore the rest in this function

Reply:
Good point, and nice idea about the metrics - I think the histogram will be useful - and we can also use it to identify number of supernodes vs full nodes.

Reply:
Add int gauge here https://github.com/sigp/lighthouse/pull/6308/commits/d82c806accd87025672bb7c7664491828702088e

Reply:
The `peers_per_custody_subnet_count` metrics has been added, and can be used to track peer custody subnet counts, number of supernodes, and number of peerdas ready nodes